### PR TITLE
Show consumergroup in describe topic

### PR DESCRIPTION
**Pull request by hfjn**

### CODE REVIEW

File: esque/cli/commands.py [hunk 1]
```diff
@@ -223,9 +224,18 @@ def apply(state: State, file: str):
...
+    required=False,
+    is_flag=True,
+    default=False,
+    help=f"Will output the consumergroups reading from this topic. "
+    f"{red_bold('Beware! This can be a really expensive operation.')}",
+)
 @error_handler
 @pass_state
-def describe_topic(state, topic_name):
+def describe_topic(state, topic_name, consumers):
```
Comment (garrettthomaskth) [inline esque/cli/commands.py]: consumers: bool

File: esque/cli/commands.py [hunk 2]
```diff
@@ -236,6 +246,18 @@ def describe_topic(state, topic_name):
 
     click.echo(pretty(config))
```
Comment (garrettthomaskth) [inline esque/cli/commands.py]: Remove,

File: esque/cli/commands.py [hunk 3]
```diff
@@ -236,6 +246,18 @@ def describe_topic(state, topic_name):
...
+        consumergroup_controller = ConsumerGroupController(state.cluster)
+        groups = consumergroup_controller.list_consumer_groups()
+
+        consumergroups = [
+            group_name
+            for group_name in groups
+            if topic_name in consumergroup_controller.get_consumergroup(group_name).topics
+        ]
+
+        click.echo(pretty({"Consumergroups": consumergroups}))
```
Comment (garrettthomaskth) [inline esque/cli/commands.py]: config["consumergroups"] = consumergroups

File: esque/cli/commands.py [hunk 4]
```diff
@@ -236,6 +246,18 @@ def describe_topic(state, topic_name):
...
+
```
Comment (garrettthomaskth) [inline esque/cli/commands.py:277]: Click.echo(format_output(config, output_format)). The output_format function is in the stdin/stdout branch

File: esque/resources/consumergroup.py [hunk 5]
```diff
@@ -28,6 +29,30 @@ def _pykafka_group_coordinator(self) -> pykafka.Broker:
...
 
+    @property
+    def topics(self):
+        consumer_id = self.id.encode("UTF-8")
+
+        # Get Consumers who already have an offset
+        consumer_offsets = self._unpack_offset_response(
+            self._pykafka_group_coordinator.fetch_consumer_group_offsets(consumer_id, preqs=[])
+        )
+        topic_with_offsets = [topic.decode("UTF-8") for topic in consumer_offsets.keys()]
```
Comment (garrettthomaskth) [inline esque/resources/consumergroup.py]: Use set comprehension instead of list comprehension

File: esque/resources/consumergroup.py [hunk 6]
```diff
@@ -28,6 +29,30 @@ def _pykafka_group_coordinator(self) -> pykafka.Broker:
...
+
+        topics_with_members = []
+        # Get Consumers which have a member
+        try:
+            resp = self._pykafka_group_coordinator.describe_groups([consumer_id])
+            meta = self._unpack_consumer_group_response(resp.groups[consumer_id])
+            topics_with_members = [
```
Comment (garrettthomaskth) [inline esque/resources/consumergroup.py]: Set instead on list

File: esque/resources/consumergroup.py [hunk 7]
```diff
@@ -28,6 +29,30 @@ def _pykafka_group_coordinator(self) -> pykafka.Broker:
...
+                member["member_metadata"]["subscription"][0].decode("UTF-8") for member in meta["members"]
+            ]
+        except struct.error:
+            pass
+
+        # Deduplicate for Consumergroups which have offsets _and_ active members
+        return list(set(topic_with_offsets + topics_with_members))
```
Comment (garrettthomaskth) [inline esque/resources/consumergroup.py]: List(topic_with_offsets.update(topics_with_members))

File: tests/integration/commands/test_describe.py [hunk 8]
```diff
@@ -9,3 +9,15 @@ def test_smoke_test_describe_topic(cli_runner: CliRunner, topic: str):
...
+
+@pytest.mark.integration
+def test_describe_topic_consumergroup_in_output(
+    cli_runner: CliRunner, filled_topic: str, partly_read_consumer_group: str
+):
+    result = cli_runner.invoke(describe_topic, [filled_topic, "-C"])
+
+    assert result.exit_code == 0
+    # TODO: Would like to be able to read this as yaml. We should implement json and yaml outputs
+    # output = yaml.safe_load(result.output)
```
Comment (garrettthomaskth) [inline tests/integration/commands/test_describe.py]: We could do this after the stdin/stdout branch is merged. Let's look at this together tomorrow:)

File: tests/integration/commands/test_describe.py [hunk 9]
```diff
@@ -109,3 +109,14 @@ def check_described_broker(described_broker: Union[str, dict]):
         keys = described_broker.keys()
         for option in VARIOUS_IMPORTANT_BROKER_OPTIONS:
             assert option in keys
+
+
+@pytest.mark.integration
```
Comment (garrettthomaskth) [inline tests/integration/commands/test_describe.py:115]: @parameterized_output_format
def test_describe_topic_consumergroup_in_output(
    non_interactive_cli_runner: CliRunner, filled_topic: str, partly_read_consumer_group: str, output_format: str,  loader: Callable
):
result = non_interactive_cli_runner.invoke(describe_topic, ["-o", output_format], filled_topic)

File: esque/resources/consumergroup.py [hunk 10]
```diff
@@ -28,6 +29,29 @@ def _pykafka_group_coordinator(self) -> pykafka.Broker:
...
+    def topics(self):
+        consumer_id = self.id.encode("UTF-8")
+
+        # Get Consumers who already have an offset
+        consumer_offsets = self._unpack_offset_response(
+            self._pykafka_group_coordinator.fetch_consumer_group_offsets(consumer_id, preqs=[])
+        )
+        topic_with_offsets = set(topic.decode("UTF-8") for topic in consumer_offsets.keys())
+
+        topics_with_members = []
```
Comment (garrettthomaskth) [inline esque/resources/consumergroup.py]: delete ```topics_with_members = []```

Comment (garrettthomaskth) [inline esque/resources/consumergroup.py]: topics_with_members = set(), or else ```TypeError: unsupported operand type(s) for |: 'set' and 'list'``` when there is a struct.error

Comment (hfjn) [reply esque/resources/consumergroup.py]: ah god damnit.

**Review by ognjen-j — APPROVED** [review]

**Pull request merged**